cmd/node-termination-handler.go

```diff
@@ -168,8 +168,9 @@ func main() {
 			err = handleRebootUncordon(nthConfig.NodeName, interruptionEventStore, *node)
 			if err != nil {
 				log.Warn().Err(err).Msgf("Unable to complete the uncordon after reboot workflow on startup, retrying")
+				return false, nil
 			}
-			return false, nil
```

Comment:
Hi @LikithaVemulapalli,

The change here to `return true` understands that `handleRebootUncordon` has not failed, because it has not returned any `err`. The `return false` is still present, but inside the `if err != nil`, so a retry is only triggered when `handleRebootUncordon` fails.

The current behavior without this PR is to always execute the function multiple times, even when it has already succeeded, and then show an error because the `PollUntilContextCancel` has never received a successful response.

Reply:
thanks for the clarification @xabinapal, approving the PR, appreciate the efforts for fixing it :)